occurrence-download/src/main/java/org/gbif/occurrence/download/file/specieslist/SpeciesListCollector.java

```diff
@@ -0,0 +1,51 @@
+package org.gbif.occurrence.download.file.specieslist;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import org.gbif.dwc.terms.GbifTerm;
+import org.gbif.occurrence.download.hive.DownloadTerms;
+
+/**
+ * 
+ * Utility class which collects distinct species info.
+ *
+ */
+public class SpeciesListCollector {
+
+  private final Map<String,Map<String,String>> distinctSpeciesRecord = new HashMap<>();
+  
+  /**
+   * @return set of records of distinct species.
+   */
+  public Set<Map<String, String>> getDistinctSpecies() {
+    return new HashSet<>(distinctSpeciesRecord.values());
+  }
+  
+  /**
+   * group results by taxon key and order them in {@link DownloadTerms} species list download order.
+   * @return distinct species
+   */
+  public void computeDistinctSpecies(Map<String, String> filteredResult) {
```

Comment:
filteredResult is confusing name for this param, record? occurrenceSpeciesRecord ? occurrenceRecord? occurrence?